list/schema/schema.go

```diff
@@ -3,5 +3,195 @@
 
 package schema
 
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/internal/fwschema"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+)
+
+// Schema must satify the fwschema.Schema interface.
+var _ fwschema.Schema = Schema{}
+
+// Schema defines the structure and value types of resource data. This type
+// is used as the resource.SchemaResponse type Schema field, which is
+// implemented by the resource.DataSource type Schema method.
 type Schema struct {
+	// Attributes is the mapping of underlying attribute names to attribute
+	// definitions.
+	//
+	// Names must only contain lowercase letters, numbers, and underscores.
+	// Names must not collide with any Blocks names.
+	Attributes map[string]Attribute
+
+	// Blocks is the mapping of underlying block names to block definitions.
+	//
+	// Names must only contain lowercase letters, numbers, and underscores.
+	// Names must not collide with any Attributes names.
+	Blocks map[string]Block
+
+	// Description is used in various tooling, like the language server, to
+	// give practitioners more information about what this resource is,
+	// what it's for, and how it should be used. It should be written as
+	// plain text, with no special formatting.
+	Description string
+
+	// MarkdownDescription is used in various tooling, like the
+	// documentation generator, to give practitioners more information
+	// about what this resource is, what it's for, and how it should be
+	// used. It should be formatted using Markdown.
+	MarkdownDescription string
+
+	// DeprecationMessage defines warning diagnostic details to display when
+	// practitioner configurations use this resource. The warning diagnostic
+	// summary is automatically set to "Resource Deprecated" along with
+	// configuration source file and line information.
+	//
+	// Set this field to a practitioner actionable message such as:
+	//
+	//  - "Use examplecloud_other resource instead. This resource
+	//    will be removed in the next major version of the provider."
+	//  - "Remove this resource as it no longer is valid and
+	//    will be removed in the next major version of the provider."
+	//
+	DeprecationMessage string
+
+	// Version indicates the current version of the resource schema. Resource
+	// schema versioning enables state upgrades in conjunction with the
+	// [resource.ResourceWithStateUpgrades] interface. Versioning is only
+	// required if there is a breaking change involving existing state data,
+	// such as changing an attribute or block type in a manner that is
+	// incompatible with the Terraform type.
+	//
+	// Versions are conventionally only incremented by one each release.
+	Version int64
```

Comment:
Are these the final attribute fields or are they carryover from resource/schema? Like I'm not sure what `Version` would be used for if there is no state representing this schema.